### PR TITLE
Fixing TPU training by disabling wandb.watch gradients logging

### PR DESCRIPTION
**Pull request by misrasaurabh1**
Fixes issue https://github.com/huggingface/transformers/issues/4814
PyTorch TPU trainer.py had a bug where the training would freeze up during the logging step. On investigation, the culprit was found to be a wandb.watch call which was trying to log gradients. This operation is suspected to be unsupported by Wandb for TPUs. Waiting for a confirmation of TPU gradient logging support by the Wandb team.

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -341,8 +341,8 @@ def _setup_wandb(self):
                 'Automatic Weights & Biases logging enabled, to disable set os.environ["WANDB_DISABLED"] = "true"'
             )
             wandb.init(project=os.getenv("WANDB_PROJECT", "huggingface"), config=vars(self.args))
-            # keep track of model topology and gradients
-            if os.getenv("WANDB_WATCH") != "false":
+            # keep track of model topology and gradients, unsupported on TPU
+            if not is_torch_tpu_available() and os.getenv("WANDB_WATCH") != "false":
```
Comment (borisdayma) [inline src/transformers/trainer.py:345]: We also noticed some issues with multiprocessing that are currently being investigated.
In the meantime could you also disable watching when `self.args.n_gpu > 1`

Comment (misrasaurabh1) [reply src/transformers/trainer.py:345]: I have runs that utilize multiple GPUs which have gradients logged into Wandb. Maybe you are facing some other issue.

Comment (borisdayma) [reply src/transformers/trainer.py:345]: Did you observe any reduced speed? When you look at your GPU usage on your runs do some of them seem underutilized?

It seems to be a common problems with pytorch hooks and distributed computing: https://discuss.pytorch.org/t/analyzing-hook-drastically-decreases-performance/53922/3

Comment (misrasaurabh1) [reply src/transformers/trainer.py:345]: Not really, the difference in utilization between GPUs is not too much. Though GPU 0 consumes more memory, maybe because it does aggregation operations.

**Review by LysandreJik — APPROVED** [review]
LGTM
